src/client/content_mapblock.cpp

```diff
@@ -774,15 +774,24 @@ void MapblockMeshGenerator::drawLiquidTop()
 	// Positive if liquid moves towards +X
 	f32 dx = (corner_levels[0][0] + corner_levels[1][0]) -
 	         (corner_levels[0][1] + corner_levels[1][1]);
-	f32 tcoord_angle = atan2(dz, dx) * core::RADTODEG;
 	v2f tcoord_center(0.5, 0.5);
 	v2f tcoord_translate(blockpos_nodes.Z + p.Z, blockpos_nodes.X + p.X);
-	tcoord_translate.rotateBy(tcoord_angle);
+	v2f dir = v2f(dx, dz).normalize();
+	if (dir == v2f{0.0f, 0.0f}) // if corners are symmetrical
+		dir = v2f{1.0f, 0.0f};
```

Comment:
This would save another sqrt and a few multiplications, if that's wanted
```suggestion
	v2f dir;
	if (dx == 0 && dz == 0)
		dir = v2f(dx, dz).normalize();
	else // if corners are symmetrical
		dir = v2f{1.0f, 0.0f};
```

Reply:
Only 2 multiplications (`normalize` performs the check before computing the sqrt) and only when dx=dz=0. So more like style preference.